pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -203,6 +205,11 @@ func (b *AutoscalingGroupModelBuilder) buildLaunchConfigurationTask(c *fi.ModelB
 		SecurityGroups:                []*awstasks.SecurityGroup{sgLink},
 	}
 
+	if ig.Spec.InstanceMetadata != nil {
+		t.HTTPTokens = ig.Spec.InstanceMetadata.HTTPTokens
+		t.HTTPPutResponseHopLimit = ig.Spec.InstanceMetadata.HTTPPutResponseHopLimit
+	}
+
```

Comment:
```suggestion
	t.HTTPTokens = fi.String("optional")
	if ig.Spec.InstanceMetadata != nil && ig.Spec.InstanceMetadata.HTTPTokens != nil {
		t.HTTPTokens = ig.Spec.InstanceMetadata.HTTPTokens
	}
	t.HTTPPutResponseHopLimit = fi.Int64(1)
	if ig.Spec.InstanceMetadata != nil && ig.Spec.InstanceMetadata.HTTPPutResponseHopLimit != nil {
		t.HTTPPutResponseHopLimit = ig.Spec.InstanceMetadata.HTTPPutResponseHopLimit
	}

```

Reply:
the aws sdk allows these values to be nil and picks up the default values from there. I think that would be simpler here?  if t.HttpPutResponseHopLimit is nil then it is defaulted to 1 and if t.HttpTokens is nil, it is defaulted to "optional" by the aws-sdk. I thought letting the aws-sdk default it would be neater?

Reply:
Problem is how kOps computes the changes on updates and removal. If expected is `nil`, it will ignore the actual value and will never unset it.

Reply:
will apply your changes to my commits. But do let me know what you think of this ^

Reply:
ahh got your point here. I didn't know this. thank you.

Reply:
If you use nil values, kOps will not compute any changes.